File: Data Structures & Algorithms/max-water-container/submission-0.go
func maxArea(heights []int) int {
	max := 0 
	for i := 0; i < len(heights); i++ {
		for j := i + 1; j < len(heights); j++ {
			area := min(heights[i], heights[j]) * (j - i) 
			if area > max {
				max = area
			} 
		}
	}

	return max
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        st = [-1]
        area = 0
        
        for i in range(len(heights)): 
            while st[-1] != -1 and heights[st[-1]] >= heights[i]:
                top = st.pop()
                height = heights[top]
                width = i - st[-1] - 1 
                area = max(area, height * width)
            st.append(i)
            
        while st[-1] != -1:
            top = st.pop()
            height = heights[top]
            width = len(heights) - st[-1] - 1 
            area = max(area, height * width)
         
        return area